### PR TITLE
Add Rust web framework Actix

### PR DESCRIPTION
**Pull request by fafhrd91**
Add Actix asynchronous web framework for Rust language.

### CODE REVIEW

File: frameworks/Rust/actix/Cargo.lock [hunk 1]
```diff
@@ -0,0 +1,1144 @@
...
+"checksum vcpkg 0.2.2 (registry+https://github.com/rust-lang/crates.io-index)" = "9e0a7d8bed3178a8fb112199d466eeca9ed09a14ba8ad67718179b4fd5487d0b"
+"checksum version_check 0.1.3 (registry+https://github.com/rust-lang/crates.io-index)" = "6b772017e347561807c1aa192438c5fd74242a670a6cffacc40f2defd1dc069d"
+"checksum void 1.0.2 (registry+https://github.com/rust-lang/crates.io-index)" = "6a02e4885ed3bc0f2de90ea6dd45ebcbb66dacffe03547fadbb0eeae2770887d"
+"checksum walkdir 1.0.7 (registry+https://github.com/rust-lang/crates.io-index)" = "bb08f9e670fab86099470b97cd2b252d6527f0b3cc1401acdb595ffc9dd288ff"
+"checksum winapi 0.2.8 (registry+https://github.com/rust-lang/crates.io-index)" = "167dc9d6949a9b857f3451275e911c3f44255842c1f7a76f33c55103a909087a"
+"checksum winapi-build 0.1.1 (registry+https://github.com/rust-lang/crates.io-index)" = "2d315eee3b34aca4797b2da6b13ed88266e6d612562a0c46390af8299fc699bc"
+"checksum ws2_32-sys 0.2.1 (registry+https://github.com/rust-lang/crates.io-index)" = "d59cefebd0c892fa2dd6de581e937301d8552cb44489cdff035c6187cb63fa5e"
```
Comment (msmith-techempower) [inline frameworks/Rust/actix/Cargo.lock]: Please tell me this file is generated.

File: frameworks/Rust/actix/benchmark_config.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      "query_url": "/queries?q=",
+      "update_url": "/updates?q=",
+      "port": 8080,
+      "approach": "Realistic",
+      "classification": "Micro",
+      "database": "Postgres",
+      "framework": "actix",
+      "language": "Rust",
+      "orm": "Full",
+      "platform": "Rust",
```
Comment (msmith-techempower) [inline frameworks/Rust/actix/benchmark_config.json]: This should be "None"

File: frameworks/Rust/actix/src/db.rs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    type Item = Vec<models::Fortune>;
+    type Error = io::Error;
+}
+
+impl Handler<TellFortune> for DbExecutor {
+    fn handle(&mut self, _: TellFortune, _: &mut Self::Context) -> Response<Self, TellFortune>
+    {
+        use schema::Fortune::dsl::*;
+
+        match Fortune.load::<models::Fortune>(&self.0) {
```
Comment (msmith-techempower) [inline frameworks/Rust/actix/src/db.rs]: What's with all these `something.0`s? You don't have to change anything, I just want to know what it means.

File: frameworks/Rust/actix/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    HttpServer::new(
+        move || Application::with_state(State{db: addr.clone()})
+            .resource("/json", |r| r.f(json))
+            .resource("/plaintext", |r| r.f(plaintext))
+            .resource("/db", |r| r.route().a(world_row))
+            .resource("/queries", |r| r.route().a(queries))
+            .resource("/fortune", |r| r.route().a(fortune))
+            .resource("/updates", |r| r.route().a(updates)))
+        .backlog(8192)
+        .bind("127.0.0.1:8080").unwrap()
```
Comment (msmith-techempower) [inline frameworks/Rust/actix/src/main.rs]: We employ a host entry for `TFB_SERVER` (or something else) that point to the IP we use in the benchmarks and in the past some implementations have had trouble with `127.0.0.1`; is it possible to bind to something like `0.0.0.0:8080` instead? If not, have you tested with a three machine/vm setup (as is our production benchmark environment) to ensure that the client machine can actually hit this application running on the server?

File: frameworks/Rust/actix/Cargo.lock (hunk 1 above)
Comment (fafhrd91) [reply frameworks/Rust/actix/Cargo.lock]: yes, this is auto generated, but this is same as "requirements.txt" for python project. this file serve purpose of reproducible build (freezing dependencies)

File: frameworks/Rust/actix/src/db.rs (hunk 3 above)
Comment (fafhrd91) [reply frameworks/Rust/actix/src/db.rs]: in rust you can represent structs in two form. normal form with named attributes and tuple struct.
for tuple struct you address attributes by index. so following code is similar

```rust
pub struct DbExecutor(PgConnection);

fn handle(&mut self, _: RandomWorld, _: &mut Self::Context) -> Response<Self, RandomWorld>
{
        match World::load(&self.0)
        {
            Ok(mut items) => ...
            Err(_) => ...
        }
}
```

vs

```rust
pub struct DbExecutor{
   conn: PgConnection
}

fn handle(&mut self, _: RandomWorld, _: &mut Self::Context) -> Response<Self, RandomWorld>
{
        match World::load(&self.conn)
        {
            Ok(mut items) => ...
            Err(_) => ...
        }
}
```

**Pull request merged**